src/main/java/org/spongepowered/api/event/sound/PlaySoundEvent.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.sound;
+
+import org.spongepowered.api.block.tileentity.Jukebox;
+import org.spongepowered.api.effect.sound.record.RecordType;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.Event;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+
+public interface PlaySoundEvent extends Event, Cancellable {
+
+    /**
+     * Gets the location of the sound being played.
+     *
+     * @return the location of the sound being played
+     */
+    Location<World> getLocation();
+
+    /**
+     * Called when a call to World.playBroadcastSound is made, for global sounds
+     * Eg, Wither / Dragon death
+     * <p>
+     * If this event gets cancelled, it will not send the sound to the clients.
+     * </p>
+     */
+    interface Broadcast extends PlaySoundEvent {
+
+    }
+
+    /**
+     * Called when a record is inserted into a Jukebox
+     * <p>
+     * If this event gets cancelled, it will not send the sound to the clients.
+     * </p>
+     */
+    interface Record extends PlaySoundEvent {
```

Comment:
I don't think this should be here. A sound (in this context) and a record are two different things - a sound is usually a reaction to an event and is quick, while a record is background music as opposed to a quick "sound".

I think this should be a `PlayRecordEvent` or something similar that can be subinterfaced if necessary (start/stop, for example).

Then, the `SoundType` (or whatever) that the `PlaySoundEvent` represents can be moved to the `PlaySoundEvent` interface.

Reply:
I think it can still be here, it is a sound as far as the server and client are concerned, it's not like broadcast sounds are 'quick' either.

Reply:
Right, yes, reviewing the server code that makes sense. I would _much_ rather have it split out _conceptually_ but looking into the MCP implementation, and crucially, Forge, that's not going to happen.

My other thoughts are general and I will put them in the main body..